analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CastShouldNotBeDuplicated.CSharp11.cs

```diff
@@ -7,9 +7,22 @@ void ListPattern()
     {
         object[] numbers = { 1, 2, 3 };
 
-        if (numbers is [double something, 3, 3])    // FN
+        if (numbers is [Fruit fruit, 3, 3])    // Secondary
+//                      ^^^^^^^^^^^
         {
-            var ff1 = (double)something;            // FN
+            var ff1 = (Fruit)fruit;            // Noncompliant
+        }
+
+        if (numbers is [double number, 3, 3])
+        {
+            var ff2 = (double)number;          // Compliant (casting to a ValueType)
+        }
+
+        if (numbers is [1, 2, 3] anotherNumber)
+        {
+            var ff3 = (object[])anotherNumber; // FN it will probably require a rule redesign
```

Comment:
To address this FN, it will most probably require a complete redesign of the current rule. Since the case covered by this test shouldn't be very common I suggest keeping it as FN.